# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ContentIterator.h
# Commit: 154a3c5e3427
# Full Hash: 154a3c5e3427d3ab6bb1898962b1d33c9b770118
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Bug 1932150 - ContentSubtreeIterator #3: Fix a bug in ContentSubtreeIterator where the end container is in the light DOM of a shadow host r=jjaschke
#   
#   When the end container is in light DOM of a shadow host, we stop the
#   iteration after we have finished iterating the shadow root of it.
#   
# ==============================================================================

diff -r d4e14a333cbd -r 154a3c5e3427 dom/base/ContentIterator.h
--- a/dom/base/ContentIterator.h	Thu May 08 13:37:21 2025 +0000
+++ b/dom/base/ContentIterator.h	Thu May 08 13:37:22 2025 +0000
@@ -121,6 +121,7 @@
       nsINode* aNode,
       dom::AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary =
           dom::AllowRangeCrossShadowBoundary::No,
+      nsTArray<AncestorInfo>* aInclusiveAncestorsOfEndContainer = nullptr);
   static nsIContent* GetPrevSibling(
       nsINode* aNode,
       dom::AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary =
